### ISSUE TITLE
CPCv2 does not work, issue with online evaluator

### ISSUE THREAD

**Issue by MishaLaskin** [PyTorchLightning/lightning-bolts#553]
## Bug 

If you run the `cpc_module.py` file in the `cpc` folder with online evaluation, the script crashes. The reason is that in `ssl_online.py` file in the `def to_device` function, you take the input `batch` and then:

```
inputs, y = batch
x = inputs[-1]
...
return x, y
```
Since batch has shape `[B,P,C,H,W]` for CPCv2, x has shape `[P,C,H,W]`. This, of course, cannot get encoded since you've discarded the batch dimension. I believe the reason for `x = inputs[-1]` is that SimCLR returns a batch that is a list of 3 arrays with shape `[B,C,H,W]` (prob 2 lists w/ data aug images, 1 w/ original images), so by taking `inputs[-1]` your are returning one of the `[B,C,H,W]` batches.

Once I fix the online ssl evaluator, the script runs. Additionally, the training loss on CPCv2 seems really high ~125

```
Epoch 0:  38%|████████████████████████████▌                                              | 149/392 [08:10<13:19,  3.29s/it, loss=125, v_num=8, val_nce=131]
```

SimCLR has a loss of roughly 5-6 early in training. Do you know why these two CPC algos have such drastically different loss values? Since they are both optimizing NCE loss, seems like it should be similar, no?

**Comment by Borda** [comment]
@ananyahjha93 mind have look at this one?
